cmd/cluster-cloud-controller-manager-operator/main.go

```diff
@@ -127,12 +138,19 @@ func main() {
 		LeaderElectionID:        "cluster-cloud-controller-manager-leader",
 		RetryPeriod:             &retryPeriod,
 		RenewDeadline:           &renewDealine,
+		NewCache:                cache.MultiNamespacedCacheBuilder(watchingNamespaces),
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
+	infraResource, err := getInfrastructureResource(mgr.GetAPIReader())
+	if err != nil {
+		setupLog.Error(err, "unable to get platform type from infrastructure resource")
+		os.Exit(1)
+	}
```

Comment:
Out of interest, why pull this here rather than during the reconcile? Is it because we aren't caching the cluster scoped resources? If so, might want to add a comment with the motivation

Reply:
Initial motivation was about setting up predicates. Need to grab CM name from infra resource for this

https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/61/files#diff-978f2fef08d39cafbbbfe8a75f1bf39c59f7e1a7a3c7ba1f74fd4979975cd22bR181